### PR TITLE
Added NPM to Glossary and linked to mentions

### LINKED ISSUE #7282: BUG/DOC: Missing definition in docs
## Reporting a bug

- [X] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [X] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

'NPM' see [place where used](https://numba.readthedocs.io/en/stable/developer/numba-runtime.html?highlight=NPM#cooperating-with-cpython) and [search results](https://numba.readthedocs.io/en/stable/search.html?q=NPM&check_keywords=yes&area=default) is never defined.

I'm reporting it as a bug, because I don't know what this is, and I imagine many users won't either.

### PR DESCRIPTION
**Pull request by nihalshetty-boop**
## Reference an existing issue
Fixes #7282

## Description
I've added `NPM` to the definition of [nopython mode](https://numba.readthedocs.io/en/stable/glossary.html#term-nopython-mode) in `glossary.rst`. Also linked mentions of `NPM` in three places, having five occurences in total. Thanks a lot to @HPLegion and @stuartarchibald for helping me with this issue! 

- I've read the [Contributing Guide](https://numba.pydata.org/numba-doc/dev/developer/contributing.html) before working on this.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the patch @nihalshetty-boop, it's queued for review.

**Comment by HPLegion** [comment]
Glad I could help :-)

Rendered docs seem to behave as expected: https://numba--7283.org.readthedocs.build/en/7283/developer/numba-runtime.html?highlight=npm

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. I think the proposed changes look good, but please could the lines be wrapped to 80 chars? Many thanks.

**Comment by nihalshetty-boop** [comment]
>but please could the lines be wrapped to 80 chars? 

I'm sorry I didn't quite understand, can you elaborate on this?

**Comment by nihalshetty-boop** [comment]
> > but please could the lines be wrapped to 80 chars?
> 
> @stuartarchibald I'm sorry I didn't quite understand, can you elaborate on this? Should I decrease the character count of the PR Description to 80 characters?

Understood the issue and fixed in commit `Wrapped Line Characters to <= 80`.

File: docs/source/developer/numba-runtime.rst
Comment (HPLegion) [inline docs/source/developer/numba-runtime.rst:188]: When you fixed the line wrapping here, you accidentally broke the indentation. That is why one of the tests failed. (You can also see the effect of this by watching the "Github rendered" version or the read the docs build).

I suggest you click your way through the failed test pipeline to see how the output can help with finding what's broken :-)

**Review by HPLegion — CHANGES_REQUESTED** [review]

**Comment by HPLegion** [comment]
(Right now the latest round of failed tests is shown in the bottom because you have not pushed any new commits yet. In general you can inspect previous test runs by clicking on the small red cross :x: or green check mark :heavy_check_mark: next to the fingerprint of the commit)

**Comment by nihalshetty-boop** [comment]
Great, thank you for helping me out, I've corrected the indentation now :)

**Review by HPLegion — APPROVED** [review]
This looks good to me.
- I have manually verified that the new `NPM` cross references are working on ReadTheDocs.
- The indentation error is fixed and the respective part now renders correctly

There are a couple of cosmetic changes that could be made to this file (line length violations, double spaces) But in my eyes they are not really a part of this PR, so I would personally avoid mixing them in here.

**Comment by nihalshetty-boop** [comment]
@HPLegion Awesome! If an issue could be raised about
>There are a couple of cosmetic changes that could be made to this file (line length violations, double spaces)

I would be interested in working on it. I'm not sure if I can raise issues as a first time contributor.

**Comment by HPLegion** [comment]
> I would be interested in working on it. I'm not sure if I can raise issues as a first time contributor.

Anyone can - and should - open issues if they see any :-)
For a maintenance issue like this, where you already know what the problem is, an issue is probably not even needed. You can just fix it and make a PR explaining what you did and why (in my opinion).

That said, I'd wait for feedback from a core maintainer here, to see how they feel about this. Maybe they actually prefer if those changes are integrated here.

Secondly, if you start working on this now, before this PR is merged into master, then you will likely run into merge conflicts with your cosmetics branch later, because some of the changes will touch on the same lines as the one in this PR. If you see that as an annoyance or as a possibility to learn dealing with conflicts, is up to you to decide ;-)

**Comment by nihalshetty-boop** [comment]
Yes, it's better to wait till this PR has been merged 👍

**Comment by stuartarchibald** [comment]
@HPLegion many thanks for providing guidance throughout this PR and for reviewing it, it's much appreciated.

@nihalshetty-boop, @HPLegion , I'd say it's fine to just open PRs for cosmetic fixes to e.g. docs that are quick to do and unlikely to be controversial. It's entirely fine to open PRs for anything, but obviously if there's a lot of work involved it's generally a good idea to open an issue about it first so as to gather feedback prior to starting the work to ensure that the PR has a good chance of being merged etc.

RE additional changes. To save merge conflict issues I'd either wait for this PR to be merged (probably in the merge window later today) or to base a new PR off this branch and then when this patch is merged git/github will just figure out the diff.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch.

**Comment by stuartarchibald** [comment]
@nihalshetty-boop Congratulations on your first contribution to Numba!

**Comment by nihalshetty-boop** [comment]
Thank you!

**Pull request merged**
